src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -270,25 +273,50 @@
   (connector* server config nil))
 
 (schema/defn ^:always-validate
-  queue-thread-pool :- QueuedThreadPool
-  [max-threads :- schema/Int
-   queue-max-size :- schema/Int]
-  (let [queue-min-size (min config/default-queue-min-threads queue-max-size)]
-    (QueuedThreadPool. max-threads
-                       queue-min-size
-                       config/default-queue-idle-timeout
-                       (BlockingArrayQueue.
-                         queue-min-size
-                         queue-min-size
-                         queue-max-size))))
+  queue-thread-pool :- (schema/maybe QueuedThreadPool)
+  [max-threads :- (schema/maybe schema/Int)
+   queue-max-size :- (schema/maybe schema/Int)]
+  (if (or max-threads queue-max-size)
+    (let [thread-pool (if max-threads
+                        (QueuedThreadPool. max-threads)
+                        (QueuedThreadPool.))]
+      (if queue-max-size
+        ;; The code below is definitely not ideal, but there isn't a way to set
+        ;; the maximum capacity of the QueuedThreadPool's BlockingArrayQueue
+        ;; after construction.  We're trying to avoid hard-coding our own
+        ;; defaults for other settings that we want Jetty to control, e.g., the
+        ;; initial capacity of the queue and minimum number of threads.  By
+        ;; reconstructing the QueuedThreadPool here, we can use Jetty's defaults
+        ;; for settings unrelated to `queue-max-size`.  QueuedThreadPool and
+        ;; BlockingArrayQueue construction isn't too expensive.  It mostly
+        ;; involves some initial memory allocations.  The more expensive work -
+        ;; where threads are actually started and the queue expands to fulfill
+        ;; new requests - would only happen after the QueuedThreadPool were
+        ;; started.  That won't happen for the `thread-pool` instance from
+        ;; above, which just gets thrown away as this function falls out of
+        ;; scope without ever having been started.  Also, this function would
+        ;; likely only be called once per server startup where a
+        ;; `queue-max-size` were configured.
+        (let [min-threads    (.getMinThreads thread-pool)
+              queue-min-size (min queue-max-size min-threads)]
+          (QueuedThreadPool. (.getMaxThreads thread-pool)
+                             min-threads
+                             (.getIdleTimeout thread-pool)
+                             (BlockingArrayQueue.
+                               queue-min-size
+                               queue-min-size
+                               queue-max-size)))
```

Comment:
If we go with the `BlockingArrayQueue` constructor that just takes a max capacity, we'd end up with a fixed-sized queue that's pre-allocated up to the maximum capacity specified.  See https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java#L101.

The Server's default `QueuedThreadPool`, however, creates a `BlockingArrayQueue` that is allocated with a minimum capacity with the ability to grow up toward the maximum.  See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#L92-L96.  I was thinking we'd want to preserve the "grow" capability to keep memory usage in check and behave more like the way the server would by default.

`QueuedThreadPool` sets the min size and growth rate based on the maximum of min threads in the thread pool or "8".  The "8" here also corresponds to the default value for min threads in the pool.  Since we don't allow for min threads in the pool to be configured, I just went with fixing this to min threads in the pool only.

The `(min queue-max-size min-threads)` is something Jetty doesn't protect for but that seemed worthwhile.  `BlockingArrayQueue` will throw an IllegalArgumentException if the initial capacity is larger than the maximum capacity - see https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java#L133.  Since we don't expose a way to configure initial capacity, it seemed reasonable to me to just head off the possibility of the IllegalArgumentException by adjusting the initial capacity to be equal to maximum capacity.  I suspect this to be pretty unlikely anyway to set such a small max capacity for the queue for anything but testing.